tool/src/js/converter.rs

```diff
@@ -707,9 +707,13 @@ impl<'tcx> TyGenContext<'_, 'tcx> {
                 if let Some(hir::MaybeStatic::Static) = slice.lifetime() {
                     panic!("'static not supported for JS backend.")
                 } else {
-                    let alloc = alloc.expect(
+                    let alloc = if slice.lifetime().is_none() {
+                        "diplomatRuntime.OwnedSliceLeaker"
+                    } else {
+                        alloc.expect(
                         "Must provide some allocation anchor for slice conversion generation!",
-                    );
+                    )
```

Comment:
how does fmt not fail?

Reply:
https://github.com/rust-diplomat/diplomat/pull/893